### PR TITLE
Yeoman generator for frontend project

### PR DESCRIPTION
**Pull request by TiddoLangerak**
**Do not merge** (still testing it out a bit)

Features:
- A `package.json` file.
- Babel compilation step (ES6) via `npm run build`
- Creates an `index.js` that imports the babel runtime (for some ES6 features)
- Unit tests via `Karma` and `Jasmine` (`npm run test`)
- Unit tests for Jenkins via `Karma` and `Jasmine` (`npm run jenkins-test`)
- Default test to check if the bundle can be loaded
- [Magnet-test-utils](https://github.com/Magnetme/magnet-frontend-test-utils) loaded
- Configures the `package.json` to only publish to private registries (optional, asked during setup).
- Git repo with all relevant files already added.
- ng-docs (`npm run live-docs`, updates automatically)
- `.jshintrc` to instruct editors to accepts esnext.

Usage:
`yo magnet-frontend [project-name]`

Note that you must have yeoman installed for this to work, and for now also this project must be linked since it hasn't been published yet.

### CODE REVIEW

**Comment by TiddoLangerak** [comment]
@Magnetme/developers - RFR

**Comment by alexnederlof** [comment]
Does this generate what we use in pulse? Using the latest webpack etc? Why did you build this anyway?

**Comment by TiddoLangerak** [comment]
I was working on the `on-enter-viewport` thingy from Wouters PR, which had to be moved out of the Magnet.me repo (it has nothing to do with Magnet.me). I could hack the magnet-pulse setup into that repo directly, but that would still require quite a bit of manual fine tuning and stuff and still can't be reused. Since we were planning on this anyway I directly implemented it as a yeoman generator such that we can use this for all our projects in the future.

**Comment by TiddoLangerak** [comment]
And yes, this generates what we use in pulse + some extras.

**Comment by TiddoLangerak** [comment]
TODO:
- [ ] Add (empty?) `.npmignore`. Currently npm uses the `.gitignore`, which causes the `out/` folder not to be published. That makes it useless.

**Comment by alexnederlof** [comment]
awesome.

**Comment by TiddoLangerak** [comment]
I've added a shitton of stuff to it for more support of everything. Among things:
- I differentiate between *modules* and *applications*. Applications are build with `webpack`, modules are only compiled with ES6. It's up to the application that uses it to compile it into a bundle.
- Modules need standalone bundles for testing and docs, these are compiled with webpack
- Automagically created readme

**Comment by TiddoLangerak** [comment]
RFR

**Comment by TiddoLangerak** [comment]
I think that the breaking build with source maps didn't really have anything to do with the magnet.me repo, it's all working now without having changed anything there. @Magnetme/developers - Please review such that I can merge & publish!

File: src/generators/app/index.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
+import 'babel/polyfill';
+import generators from 'yeoman-generator';
+import { basename, resolve, dirname } from 'path';
+import mothership from 'mothership';
+import { readFile, exists } from 'fs';
+import glob from 'glob';
+
+export default generators.Base.extend({
```
Comment (TiddoLangerak) [inline src/generators/app/index.js]: Small explanation for this file: all keys in the object I define here below are phases for yeoman and are executed in a specific order (coincidentally the order that I've defined them in). These keys define new objects in this case, all functions in that object will be executed for that phase (in no specific order).

**Comment by alexnederlof** [comment]
:+1: muchos awesome!

**Pull request merged**